eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -0,0 +1,226 @@
+  default: $(Agent.BuildDirectory)/vmr
+
+- name: vmrBranch
+  displayName: dotnet/dotnet branch to use
+  type: string
+  default: $(Build.SourceBranchName)
+
+# Resource id of the installer build to retrieve MSFT SDK from
+- name: installerBuildResourceId
+  type: string
+
+- name: buildName
+  type: string
+
+- name: architecture
+  type: string
+
+- name: container
+  type: string
+
+- name: pool
+  type: object
+
+# Allow downloading artifacts from the internet during the build
+- name: runOnline
+  type: boolean
+
+# Skip running the SDK content smoke-tests
+- name: excludeSdkContentTests
+  type: boolean
+  default: false
+
+# Name of a previous job (from the same template as this) whose output will be used to build this job
+# The SDK from its artifacts is copied to vmr/.dotnet
+- name: reuseBuildArtifactsFrom
+  type: string
+  default: ''
+
+- name: excludeOmniSharpTests
+  type: boolean
+
+- name: bootstrapPrep
+  type: boolean
+
+- name: enablePoison
+  type: boolean
+
+jobs:
+- job: Build_${{ parameters.buildName }}_${{ parameters.architecture }}
+  timeoutInMinutes: 150
+  pool: ${{ parameters.pool }}
+  ${{ if ne(parameters.reuseBuildArtifactsFrom, '') }}:
+    # Always attempt to run the bootstrap leg (e.g. even when stage 1 tests fail) in order to get a complete accessment of the build status.
+    # The build shortcuts when stage 1 build fails and doesn't produce the SDK.
+    condition: succeededOrFailed()
+    dependsOn: Build_${{ parameters.reuseBuildArtifactsFrom }}_${{ parameters.architecture }}
+  variables:
+  - template: /eng/common/templates/variables/pool-providers.yml
+  - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+    - group: DotNetBot-GitHub
+    - group: AzureDevOps-Artifact-Feeds-Pats
+  - ${{ else }}:
+    - name: BotAccount-dotnet-bot-repo-PAT
+      value: N/A
+  - ${{ if eq(parameters.reuseBuildArtifactsFrom, '') }}:
+    - name: additionalBuildArgs
+      value: ''
+  - ${{ else }}:
+    - name: additionalBuildArgs
+      value: ' --with-sdk /vmr/.dotnet'
+
+  steps:
+  - template: ../steps/vmr-prepare.yml
+    parameters:
+      vmrBranch: ${{ parameters.vmrBranch }}
+      installerBuildResourceId: ${{ parameters.installerBuildResourceId }}
+      skipComponentGovernanceDetection: true
+
+  # Synchronize new content in the VMR during PRs (we expect this to come
+  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    - template: ../steps/vmr-pull-updates.yml
+      parameters:
+        vmrPath: ${{ parameters.vmrPath }}
+        vmrBranch: ${{ parameters.vmrBranch }}
+        targetRef: $(Build.SourceVersion) # Synchronize the current installer commit
+
+  - ${{ if ne(variables['System.TeamProject'], 'public') }}:
+    - task: Bash@3
+      displayName: Setup Private Feeds Credentials
+      inputs:
+        filePath: ${{ parameters.vmrPath }}/src/installer/eng/common/SetupNugetSources.sh
+        arguments: ${{ parameters.vmrPath }}/src/installer/NuGet.config $Token
```

Comment:
This logic is incorrect and needs changes from https://github.com/dotnet/installer/pull/15007.

This logic injects credentials that are needed for the smoke-tests.  The problem is this logic modified the installer's NuGet.config.  When source-build runs offline, it removes all sources from the nuget config including the ones this step adds.  This will break the smoke-tests.  This is only an issue for internal servicing builds which is why it doesn't manifest in PR.

From https://github.com/dotnet/installer/pull/15007 you need two things.

1. insert this step above this step:

  - script: cp ${{ parameters.vmrPath }}/src/installer/NuGet.config ${{ parameters.vmrPath }}/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/online.NuGet.Config
    displayName: Copy Test NuGet Config
    
 2. Refactor the source path in this step to operate of the online.NuGet.Config copied earlier.
 3. Remove the copy nuget config logic in the run tests step - https://github.com/dotnet/installer/pull/15042/files#diff-dc549eeb99aa73960e74c9d4c160a3c76e3e414398bb348c1d5365aac0369ad5R162

Reply:
Done in https://github.com/dotnet/installer/pull/15042/commits/26d3bee38984c6eaf49513e5e0cae268a25835cf